neo/Network/P2P/TaskManager.cs

```diff
@@ -221,7 +222,11 @@ private void RequestTasks(TaskSession session)
                     return;
                 }
             }
-            if ((!HasHeaderTask || globalTasks[HeaderTaskHash] < MaxConncurrentTasks) && Blockchain.Singleton.HeaderHeight < session.StartHeight)
+            if ((!HasHeaderTask || globalTasks[HeaderTaskHash] < MaxConncurrentTasks)
+                && (Blockchain.Singleton.HeaderHeight < session.StartHeight
```

Comment:
```if ((!HasHeaderTask || globalTasks[HeaderTaskHash] < MaxConncurrentTasks)
                && (Blockchain.Singleton.HeaderHeight < session.StartHeight
                    || (Blockchain.Singleton.HeaderHeight == session.StartHeight
                        && Blockchain.Singleton.HeaderHeight == sessions.Select(x => x.Value.StartHeight).Max()
                        && TimeProvider.Current.UtcNow.ToTimestamp() - 60 >= Blockchain.Singleton.GetBlock(Blockchain.Singleton.CurrentHeaderHash)?.Timestamp)
                        ))
```
?

Reply:
All the condition

```Blockchain.Singleton.HeaderHeight == session.StartHeight
                        && Blockchain.Singleton.HeaderHeight == sessions.Select(x => x.Value.StartHeight).Max()
                        && TimeProvider.Current.UtcNow.ToTimestamp() - 60 >= Blockchain.Singleton.GetBlock(Blockchain.Singleton.CurrentHeaderHash)?.Timestamp)
``` 

should be inside parentheses, right?

Reply:
Yes, the logic is inside parentheses.